palette/src/rgb/packed.rs

```diff
@@ -0,0 +1,305 @@
+/// RGBA color.
+///
+/// Note that conversion from float to integer component types in `palette` maps
+/// the floating point value to the integer's value range and then rounds the
+/// result before casting to the integer type. An example of the difference in
+/// conversion methods is shown below.
+///
+/// ```
+/// use approx::assert_relative_eq;
+/// use palette::{Packed, Srgb, Srgba};
+/// use palette::rgb::channels::{Argb, Rgba};
+///
+/// let packed: Packed = Srgb::new(0.5, 0.0, 0.5).into_format().into();
+/// assert_eq!(0xFF80_0080, packed.color);
+///
+/// let unpacked: Srgba<u8> = Packed::<Rgba>::from(0x80FF_FF80).into();
+/// assert_relative_eq!(
+///     Srgba::new(0.5, 1.0, 1.0, 0.5),
+///     unpacked.into_format(),
+///     epsilon = 0.01
+/// );
+///
+/// // By default, `Packed` uses `Argb` order to make entering 6-digit hex
+/// // numbers more convenient
+/// let unpacked = Srgb::<u8>::from_u32::<Argb>(0xFF8000);
+/// assert_eq!(Srgb::<u8>::new(0xFF, 0x80, 0x00), unpacked);
+///
+/// // The second assert is essentially how library components are converted
+/// let float = 0.5f32 * 255.0;
+/// assert_eq!(0x7F, float as u32);
+/// assert_eq!(0x80, float.round() as u32);
+/// ```
+///
+/// When an `Rgb` type is packed, the alpha value will be `0xFF` in the
+/// corresponding `u32`. Converting from a packed color type back to an `Rgb`
+/// type will disregard the alpha value.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Pixel)]
+#[palette_internal]
+#[repr(C)]
+pub struct Packed<C: RgbChannels = channels::Argb> {
+    /// The sRGB color packed into a `u32`.
+    pub color: u32,
+
+    /// The channel ordering for red, green, blue, and alpha components in the
+    /// packed integer; can be `Abgr`, `Argb`, `Bgra`, or `Rgba`. See
+    /// [RgbChannels](trait.RgbChannels.html).
+    #[palette_unsafe_zero_sized]
+    pub channel_order: PhantomData<C>,
+}
+
+/// Splits and combines RGB(A) types with some channel ordering. Channels may be
+/// ordered as `Abgr`, `Argb`, `Bgra`, or `Rgba`.
+pub trait RgbChannels {
+    /// Split RGBA components into a `(u8, u8, u8, u8)` tuple.
+    fn split_rgb<S: RgbStandard>(rgb: Rgba<S, u8>) -> (u8, u8, u8, u8);
+    /// Create an RGBA color from a `(u8, u8, u8, u8)` tuple.
+    fn combine_rgb<S: RgbStandard>(channels: (u8, u8, u8, u8)) -> Rgba<S, u8>;
+}
+
+impl<C: RgbChannels> From<u32> for Packed<C> {
+    fn from(color: u32) -> Self {
+        Packed {
+            color,
+            channel_order: PhantomData,
+        }
+    }
+}
+
+impl<C: RgbChannels> Into<u32> for Packed<C> {
+    fn into(self) -> u32 {
+        self.color
+    }
+}
+
+impl<S, C> Into<Rgb<S, u8>> for Packed<C>
+where
+    S: RgbStandard,
+    C: RgbChannels,
+{
+    fn into(self) -> Rgb<S, u8> {
+        let bytes: [u8; 4] = self.color.to_be_bytes();
+        let color: Alpha<Rgb<S, u8>, u8> = C::combine_rgb((bytes[0], bytes[1], bytes[2], bytes[3]));
+        Rgb::from_components((color.red, color.green, color.blue))
```

Comment:
When this is changed to be a `From` implementation, you can change it to `Rgba::from(packed).color`. `Alpha<Rgb>` wraps an `Rgb` as `color` and exposes its content via `Deref`, so that's it!